specifications/17-dependent-features.json

```diff
@@ -0,0 +1,738 @@
+        "dependencies": [{
+          "feature": "parent.not.exist"
+        }],
+        "strategies": [{
+          "name": "flexibleRollout",
+          "parameters": {
+            "rollout": "100",
+            "stickiness": "default",
+            "groupId": "groupId"
+          },
+          "variants": [
+            {
+              "name": "child.variant",
+              "weight": 1,
+              "payload": {
+                "type": "string",
+                "value": "variantValue"
+              }
+            }
+          ]
+        }]
+      }
+    ]
+  },
+  "tests": [
+    {
+      "description": "Child and parent enabled",
+      "context": {},
+      "toggleName": "parent.enabled.child.enabled",
+      "expectedResult": true
+    },{
+      "description": "Child disabled",
+      "context": {},
+      "toggleName": "parent.enabled.child.disabled",
+      "expectedResult": false
+    },
+    {
+      "description": "Parent disabled",
+      "context": {},
+      "toggleName": "parent.disabled.child.enabled",
+      "expectedResult": false
+    },
+    {
+      "description": "Child and parent disabled",
+      "context": {},
+      "toggleName": "parent.disabled.child.disabled",
+      "expectedResult": false
+    },
+    {
+      "description": "Parent dependency expects disabled",
+      "context": {},
+      "toggleName": "parent.disabled.satisfied.child.enabled",
+      "expectedResult": true
+    },
+    {
+      "description": "Parent dependency disabled expectation not satisfied",
+      "context": {},
+      "toggleName": "parent.disabled.not.satisfied.child.enabled",
+      "expectedResult": false
+    },
+    {
+      "description": "Parent dependency with variant",
+      "context": {},
+      "toggleName": "parent.single.variant.child.enabled",
+      "expectedResult": true
+    },
+    {
+      "description": "Parent dependency with one of variants satisfied",
+      "context": {},
+      "toggleName": "parent.multiple.variants.child.enabled",
+      "expectedResult": true
+    },
+    {
+      "description": "Parent dependency with empty variants",
+      "context": {},
+      "toggleName": "parent.empty.variants.child.enabled",
+      "expectedResult": true
+    },
+    {
+      "description": "Parent dependency with non matching variant",
+      "context": {},
+      "toggleName": "parent.non.matching.variant.child.enabled",
+      "expectedResult": false
+    },
+    {
+      "description": "Multiple parent dependencies satisfied",
+      "context": {},
+      "toggleName": "multiple.parents.satisfied.child.enabled",
+      "expectedResult": true
+    },
+    {
+      "description": "Multiple parents not satisfied",
+      "context": {},
+      "toggleName": "multiple.parents.not.satisfied.child.enabled",
+      "expectedResult": false
+    },
+    {
+      "description": "Parent does not exist",
+      "context": {},
+      "toggleName": "parents.not.exist.child.enabled",
+      "expectedResult": false
```

Comment:
Nice, that's what I would have done too, I think this is correct. This is a big decision though. It means stale caches and/or cycling out your api key can potentially change the state of a toggle. Leaving this here for visibility in case someone has a different opinion

Reply:
Glad you're confirming my confirmation bias :) As you said it's up for a discussion.

Reply:
This is tricky. Since you can filter by tags it's technically possible to end up in a situation where the parent is not present. We'll need to implement safeguards in the SDK to warn you if this happens, but I think the behavior here is correct. If you depend on a feature toggle that you have not fetched, there is no way for us to know the evaluation result of that toggle and we can not determine that the child should be enabled.

Reply:
I guess this is where a feature toggle type would come in handy @kwasniew

Reply:
@FredrikOseberg I will try to add a warning when you try to depend on a feature that you don't have in the SDK

Reply:
@FredrikOseberg can you elaborate on a feature toggle type?

Reply:
Yes. Like we talked about in the meeting, some way to mark a toggle as dependable, we discussed the possibility of having a new feature toggle type called Dependable Toggle. If we had that, we could circumvent the filtering in the API and say that "No, we don't care that you are filtering by tags, you'll get all the dependable toggles in this project regardless".

Reply:
@FredrikOseberg added reference impl for warnings: https://github.com/Unleash/unleash-client-node/pull/520/commits/1ed0994bb2c598e8289320d3a0b4243a90e4a785

Reply:
Now I remember the dependent feature markers we talked about :) We definitely need to keep it in mind. For this spec though I think we can proceed without this concept since it will only affect the client API not this spec.

Reply:
Agree. This spec just surfaced the problem, so I just mentioned it here.